apps/studio/components/interfaces/SignIn/SignInForm.tsx

```diff
@@ -1,34 +1,44 @@
-import HCaptcha from '@hcaptcha/react-hcaptcha'
+import { useRef, useState, useEffect } from 'react'
+import { useRouter } from 'next/router'
+import Link from 'next/link'
+import { useQueryClient } from '@tanstack/react-query'
+import { type SubmitHandler, useForm } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import z from 'zod'
 import * as Sentry from '@sentry/nextjs'
 import type { AuthError } from '@supabase/supabase-js'
-import { useQueryClient } from '@tanstack/react-query'
-import Link from 'next/link'
-import { useRouter } from 'next/router'
-import { useRef, useState, useEffect } from 'react'
+import HCaptcha from '@hcaptcha/react-hcaptcha'
 import { toast } from 'sonner'
-import { object, string } from 'yup'
 
 import { useAddLoginEvent } from 'data/misc/audit-login-mutation'
 import { getMfaAuthenticatorAssuranceLevel } from 'data/profile/mfa-authenticator-assurance-level-query'
 import { useSendEventMutation } from 'data/telemetry/send-event-mutation'
 import { useLastSignIn } from 'hooks/misc/useLastSignIn'
 import { auth, buildPathWithParams, getReturnToPath } from 'lib/gotrue'
-import { Button, Form, Input } from 'ui'
+import { Button, Form_Shadcn_, FormControl_Shadcn_, FormField_Shadcn_, Input_Shadcn_ } from 'ui'
 import { LastSignInWrapper } from './LastSignInWrapper'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
 
-const signInSchema = object({
-  email: string().email('Must be a valid email').required('Email is required'),
-  password: string().required('Password is required'),
+const schema = z.object({
+  email: z.string({ required_error: 'Email is required' }).email('Must be a valid email'),
+  password: z.string({ required_error: 'Password is required' }),
```

Comment:
these validations don't seem to be working - able to submit the form with 0 characters for password

only the email validation seems to be working (RE Must be a valid email)

Reply:
will need to use `.min()` instead of `required_error`